Error while checking build/O2/fullCI for f7d4bb472eb427f66dd1aebd9e3ce39ff0e81a03 at 2024-08-14 12:55:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/13383-slc8_x86-64/0/GPU/GPUTracking/Global/GPUChainITS.cxx:40:14: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
/sw/BUILD/7eba8e02fc26d8cf819e2beec23b8dfbb18a9b4c/O2/Detectors/ITSMFT/ITS/tracking/GPU/hip/VertexerTraitsGPU.hip:94:20: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13383/f7d4bb472eb427f66dd1aebd9e3ce39ff0e81a03/build_O2_fullCI/pretty.html).

--- a/Detectors/ITSMFT/ITS/tracking/GPU/ITStrackingGPU/VertexerTraitsGPU.h
+++ b/Detectors/ITSMFT/ITS/tracking/GPU/ITStrackingGPU/VertexerTraitsGPU.h
@@ -41,21 +41,20 @@ class VertexerTraitsGPU : public VertexerTraits
 {
  public:
   VertexerTraitsGPU();
-  ~VertexerTraitsGPU() override;
+  ~VertexerTraitsGPU() = default;
   void initialise(const TrackingParameters&, const int iteration = 0) override;
   void adoptTimeFrame(TimeFrame*) override;
   void computeTracklets(const int iteration = 0) override;
   void computeTrackletMatching(const int iteration = 0) override;
   void computeVertices(const int iteration = 0) override;
+  void updateVertexingParameters(const std::vector<VertexingParameters>&, const TimeFrameGPUParameters&) override;
 
   // Hybrid
-  void initialiseHybrid(const TrackingParameters& pars) override { VertexerTraits::initialise(pars); }
+  void initialiseHybrid(const TrackingParameters& pars, const int iteration = 0) override { VertexerTraits::initialise(pars, iteration); }
+  void computeTrackletsHybrid(const int iteration = 0) override { VertexerTraits::computeTracklets(iteration); }
+  void computeTrackletMatchingHybrid(const int iteration = 0) override { VertexerTraits::computeTrackletMatching(iteration); }
+  void computeVerticesHybrid(const int iteration = 0) override { VertexerTraits::computeVertices(iteration); }
   void adoptTimeFrameHybrid(TimeFrame* tf) override { VertexerTraits::adoptTimeFrame(tf); }
-  void computeTrackletsHybrid() override { VertexerTraits::computeTracklets(); }
-  void computeTrackletMatchingHybrid() override { VertexerTraits::computeTrackletMatching(); }
-  void computeVerticesHybrid() override { VertexerTraits::computeVertices(); }
-
-  void updateVertexingParameters(const std::vector<VertexingParameters>&, const TimeFrameGPUParameters&) override;
 
   void computeVerticesHist();
 

--- a/Detectors/ITSMFT/ITS/tracking/GPU/cuda/VertexerTraitsGPU.cu
+++ b/Detectors/ITSMFT/ITS/tracking/GPU/cuda/VertexerTraitsGPU.cu
@@ -91,10 +91,6 @@ VertexerTraitsGPU::VertexerTraitsGPU()
   setIsGPU(true);
 }
 
-VertexerTraitsGPU::~VertexerTraitsGPU()
-{
-}
-
 void VertexerTraitsGPU::initialise(const TrackingParameters& trackingParams, const int iteration)
 {
   mTimeFrameGPU->initialise(0, trackingParams, 3, &mIndexTableUtils, &mTfGPUParams);

--- a/Detectors/ITSMFT/ITS/tracking/include/ITStracking/Vertexer.h
+++ b/Detectors/ITSMFT/ITS/tracking/include/ITStracking/Vertexer.h
@@ -72,10 +72,12 @@ class Vertexer
   void findTrivialMCTracklets();
   template <typename... T>
   void validateTracklets(T&&... args);
-  void validateTrackletsHybrid();
+  template <typename... T>
+  void validateTrackletsHybrid(T&&... args);
   template <typename... T>
   void findVertices(T&&... args);
-  void findVerticesHybrid();
+  template <typename... T>
+  void findVerticesHybrid(T&&... args);
   void findHistVertices();
 
   template <typename... T>
@@ -153,14 +155,16 @@ void Vertexer::findTrackletsHybrid(T&&... args)
   mTraits->computeTrackletsHybrid(std::forward<T>(args)...);
 }
 
-inline void Vertexer::validateTrackletsHybrid()
+template <typename... T>
+inline void Vertexer::validateTrackletsHybrid(T&&... args)
 {
-  mTraits->computeTrackletMatchingHybrid();
+  mTraits->computeTrackletMatchingHybrid(std::forward<T>(args)...);
 }
 
-inline void Vertexer::findVerticesHybrid()
+template <typename... T>
+inline void Vertexer::findVerticesHybrid(T&&... args)
 {
-  mTraits->computeVerticesHybrid();
+  mTraits->computeVerticesHybrid(std::forward<T>(args)...);
 }
 
 template <typename... T>

--- a/Detectors/ITSMFT/ITS/tracking/include/ITStracking/VertexerTraits.h
+++ b/Detectors/ITSMFT/ITS/tracking/include/ITStracking/VertexerTraits.h
@@ -69,10 +69,10 @@ class VertexerTraits
   virtual void adoptTimeFrame(TimeFrame* tf);
   virtual void updateVertexingParameters(const std::vector<VertexingParameters>& vrtPar, const TimeFrameGPUParameters& gpuTfPar);
   // Hybrid
-  virtual void initialiseHybrid(const TrackingParameters& trackingParams) { initialise(trackingParams); };
-  virtual void computeTrackletsHybrid() { computeTracklets(); };
-  virtual void computeTrackletMatchingHybrid() { computeTrackletMatching(); };
-  virtual void computeVerticesHybrid() { computeVertices(); };
+  virtual void initialiseHybrid(const TrackingParameters& trackingParams, const int iteration = 0) { initialise(trackingParams, iteration); };
+  virtual void computeTrackletsHybrid(const int iteration = 0) { computeTracklets(iteration); };
+  virtual void computeTrackletMatchingHybrid(const int iteration = 0) { computeTrackletMatching(iteration); };
+  virtual void computeVerticesHybrid(const int iteration = 0) { computeVertices(iteration); };
   virtual void adoptTimeFrameHybrid(TimeFrame* tf) { adoptTimeFrame(tf); };
 
   void computeVerticesInRof(int,

--- a/Detectors/ITSMFT/ITS/tracking/src/Vertexer.cxx
+++ b/Detectors/ITSMFT/ITS/tracking/src/Vertexer.cxx
@@ -60,12 +60,17 @@ float Vertexer::clustersToVerticesHybrid(std::function<void(std::string s)> logg
 {
   float total{0.f};
   TrackingParameters trkPars;
-  trkPars.PhiBins = mTraits->getVertexingParameters()[0].PhiBins;
-  trkPars.ZBins = mTraits->getVertexingParameters()[0].ZBins;
-  total += evaluateTask(&Vertexer::initialiseVertexerHybrid, "Hybrid Vertexer initialisation", logger, trkPars);
-  total += evaluateTask(&Vertexer::findTrackletsHybrid, "Hybrid Vertexer tracklet finding", logger);
-  total += evaluateTask(&Vertexer::validateTrackletsHybrid, "Hybrid Vertexer adjacent tracklets validation", logger);
-  total += evaluateTask(&Vertexer::findVerticesHybrid, "Hybrid Vertexer vertex finding", logger);
+  TimeFrameGPUParameters tfGPUpar;
+  mTraits->updateVertexingParameters(mVertParams, tfGPUpar);
+  for (int iteration = 0; iteration < std::min(mVertParams[0].nIterations, (int)mVertParams.size()); ++iteration) {
+    logger(fmt::format("ITS Hybrid seeding vertexer iteration {} summary:", iteration));
+    trkPars.PhiBins = mTraits->getVertexingParameters()[0].PhiBins;
+    trkPars.ZBins = mTraits->getVertexingParameters()[0].ZBins;
+    total += evaluateTask(&Vertexer::initialiseVertexerHybrid, "Hybrid Vertexer initialisation", logger, trkPars, iteration);
+    total += evaluateTask(&Vertexer::findTrackletsHybrid, "Hybrid Vertexer tracklet finding", logger, iteration);
+    total += evaluateTask(&Vertexer::validateTrackletsHybrid, "Hybrid Vertexer adjacent tracklets validation", logger, iteration);
+    total += evaluateTask(&Vertexer::findVerticesHybrid, "Hybrid Vertexer vertex finding", logger, iteration);
+  }
   printEpilog(logger, total);
   return total;
 }

